core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java

```diff
@@ -129,7 +131,15 @@ public RetryPolicyTemplate getRetryPolicyTemplate() {
     return errorTypeRepository.getErrorType(getIdentifier(errorType));
   }
 
+  private Optional<ErrorType> getErrorType(org.mule.sdk.api.error.ErrorTypeDefinition errorType) {
```

Comment:
same here. The goal is to reduce all thing to a single interface. We should make sure to perform all the necessary adaptations so that our code can deal only in terms of one single interface.

Reply:
But at this point we are getting the Exception from the connector, shouldn't we deal with both interfaces here?